packages/internal-test-env/index.ts

```diff
@@ -0,0 +1,351 @@
+      );
+    }
+
+    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_REQUESTOR_CLIENT_SECRET is undefined."
+      );
+    }
+
+    if (typeof process.env.E2E_TEST_VC_PROVIDER !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_VC_PROVIDER is undefined."
+      );
+    }
+  }
+  if (
+    process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_ID !== undefined ||
+    process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_SECRET !== undefined
+  ) {
+    if (typeof process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_ID !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_RESOURCE_OWNER_CLIENT_ID is undefined."
+      );
+    }
+    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_REQUESTOR_CLIENT_SECRET is undefined."
+      );
+    }
+  }
+
+  return {
+    idp: process.env.E2E_TEST_IDP,
+    environment: process.env.E2E_TEST_ENVIRONMENT,
+    protocol: process.env.E2E_TEST_NOTIFICATION_PROTOCOL,
+    notificationGateway: process.env.E2E_TEST_NOTIFICATION_GATEWAY,
+    clientCredentials: {
+      requestor: {
+        id: process.env.E2E_TEST_REQUESTOR_CLIENT_ID,
+        secret: process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET,
+      },
+      resourceOwner: {
+        id: process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_ID,
+        secret: process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_SECRET,
+      },
+    },
+    vcProvider: process.env.E2E_TEST_VC_PROVIDER,
+    features: featuredFlags,
+  };
+}
+
+export function getBrowserTestingEnvironment(): TestingEnvironmentBrowser {
+  setupEnv();
+  getTestingEnvironment(process.env);
+
+  if (process.env.E2E_TEST_USER === undefined) {
+    throw new Error("The environment variable E2E_TEST_USER is undefined.");
+  }
+  if (process.env.E2E_TEST_PASSWORD === undefined) {
+    throw new Error("The environment variable E2E_TEST_PASSWORD is undefined.");
+  }
+
+  return {
+    clientCredentials: {
+      resourceOwner: {
+        login: process.env.E2E_TEST_USER,
+        password: process.env.E2E_TEST_PASSWORD,
+      },
+    },
+    idp: process.env.E2E_TEST_IDP,
+    notificationGateway: process.env.E2E_TEST_NOTIFICATION_GATEWAY,
+    features: featuredFlags,
+  };
+}
+
+export async function getAuthenticatedSession(
+  authDetails: TestingEnvironmentNode
+): Promise<Session> {
+  const session = new Session();
+  await session.login({
+    oidcIssuer: authDetails.idp,
+    clientId: authDetails.clientCredentials.resourceOwner.id,
+    clientSecret: authDetails.clientCredentials.resourceOwner.secret,
+  });
+
+  if (!session.info.isLoggedIn) {
+    throw new Error("Logging the test agent in failed.");
+  }
+
+  return session;
+}
+
+export async function getPodRoot(session: Session) {
+  const podRootAll = await getPodUrlAll(session.info.webId as string);
```

Comment:
Better to add a check on `session.info.webId` and throw if it's not there.

Reply:
Are you good with this comment being resolved?